plugin/src/android/buildscriptDependency.ts

```diff
@@ -0,0 +1,37 @@
+import { ConfigPlugin, withAppBuildGradle } from '@expo/config-plugins'
+
+import { PluginConfigType } from '../pluginConfig'
+
+/**
+ * Update `<project>/build.gradle` by adding the codepush.gradle file
+ * as an additional build task definition underneath react.gradle
+ */
+
+function applyImplementation(appBuildGradle: string) {
+  const codePushImplementation = `apply from: new File(["node", "--print", "require.resolve('react-native-code-push/package.json')"].execute(null, rootDir).text.trim()).getParentFile().getAbsolutePath() + "/android/codepush.gradle"`
+
+  // Make sure the project does not have the dependency already
+  if (!appBuildGradle.includes(codePushImplementation)) {
+    if (appBuildGradle.includes('apply from: new File(reactNativeRoot, "react.gradle")')) {
+      return appBuildGradle.replace(
+        'apply from: new File(reactNativeRoot, "react.gradle")',
+        `apply from: new File(reactNativeRoot, "react.gradle")\n${codePushImplementation}`
+      )
+    } else if (
+      appBuildGradle.includes('apply from: "../../node_modules/react-native/react.gradle"')
+    ) {
+      return appBuildGradle.replace(
+        'apply from: "../../node_modules/react-native/react.gradle"',
+        `apply from: "../../node_modules/react-native/react.gradle"\n${codePushImplementation}`
+      )
+    }
+  }
+
+  return appBuildGradle
```

Comment:
I'd do a little refactoring by avoiding the `else if` and replacing it with a simple if. I also suggest putting the line that will be above the line you want to add in a variable. Finally, I've added a case for react native 71 since the line containing `react.gradle` no longer exists (see https://react-native-community.github.io/upgrade-helper/?from=0.70.9&to=0.71.8#RnDiffApp-android-app-build.gradle).
```suggestion
  const codePushImplementation = `apply from: new File(["node", "--print", "require.resolve('react-native-code-push/package.json')"].execute(null, rootDir).text.trim()).getParentFile().getAbsolutePath() + "/android/codepush.gradle"`;

  // Make sure the project does not have the dependency already
  if (!appBuildGradle.includes(codePushImplementation)) {
    const reactNativeFileClassGradleInclude = `'apply from: new File(reactNativeRoot, "react.gradle")`;
    if (appBuildGradle.includes(reactNativeFileClassGradleInclude)) {
      return appBuildGradle.replace(
        reactNativeFileClassGradleInclude,
        `${reactNativeFileClassGradleInclude}\n${codePushImplementation}`
      );
    }
    const reactNativeRawGradleInclude = `apply from: "../../node_modules/react-native/react.gradle"`;
    if (appBuildGradle.includes(reactNativeRawGradleInclude)) {
      return appBuildGradle.replace(
        reactNativeRawGradleInclude,
        `${reactNativeRawGradleInclude}\n${codePushImplementation}`
      );
    }
    const reactNative71Include = `apply from: new File(["node", "--print", "require.resolve('@react-native-community/cli-platform-android/package.json')"].execute(null, rootDir).text.trim(), "../native_modules.gradle");`;
    if (appBuildGradle.includes(reactNative71Include)) {
      return appBuildGradle.replace(
        reactNative71Include,
        `${reactNative71Include}\n${codePushImplementation}`
      );
    }
  }
  return appBuildGradle;
```